okhttp-clients/src/test/java/com/palantir/remoting3/okhttp/MeshProxyInterceptorTest.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.remoting3.okhttp;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.google.common.net.HostAndPort;
+import com.google.common.net.HttpHeaders;
+import okhttp3.HttpUrl;
+import okhttp3.Interceptor;
+import okhttp3.Request;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+public final class MeshProxyInterceptorTest {
```

Comment:
I'd like to see a test that sanity-checks the interplay of this interceptors with the other interceptors that change URLs. (This is mostly my bad: the interceptor-based logic here is really a bit messed up and complicated :( )

Reply:
Hmm any suggestions on how to add this test / what it should test exactly? Considering that `ClientConfiguration` disallows retries, it effectively disables all other interceptors from rewriting URIs.

Reply:
ja, probably not much to do here. all good